src/NexusMods.App.UI/WorkspaceSystem/PanelTab/PanelTabViewModel.cs

```diff
@@ -16,9 +19,97 @@ public class PanelTabViewModel : AViewModel<IPanelTabViewModel>, IPanelTabViewMo
     /// <inheritdoc/>
     [Reactive] public bool IsVisible { get; set; } = true;
 
-    public PanelTabViewModel()
+    /// <inheritdoc/>
+    public ReactiveCommand<Unit, Unit> GoBackInHistoryCommand { get; }
+
+    /// <inheritdoc/>
+    public ReactiveCommand<Unit, Unit> GoForwardInHistoryCommand { get; }
+
+    private const int HistoryLimit = 10;
+    private readonly PageData[] _history = new PageData[HistoryLimit];
+    [Reactive] private int ReadIndex { get; set; } = -1;
+    [Reactive] private int WriteIndex { get; set; }
+    [Reactive] private int HistoryCount { get; set; }
+    private bool IsAdvancing { get; set; }
```

Comment:
I think the code below isn't the most readable, could you add some comments either here on the variables or in the code to clarify a bit the intent?

I think IsAdvancing is being used to discriminate between normal content changes and history navigation changes?
The name isn't too clear.

The looping nature of the _history buffer is also not immediately clear and HistoryCount seems nebulous